pkg/build/docker_generic.go

```diff
@@ -64,15 +61,23 @@ func (b *DockerGenericBuilder) Build(ctx context.Context, in *api.BuildInput, ow
 
 	buildStart := time.Now()
 
-	err = docker.BuildImage(ctx, ow, cli, &imageOpts)
+	imageID, err := docker.BuildImage(ctx, ow, cli, &imageOpts)
 	if err != nil {
 		return nil, fmt.Errorf("docker build failed: %w", err)
 	}
 
 	ow.Infow("build completed", "took", time.Since(buildStart).Truncate(time.Second))
 
 	out := &api.BuildOutput{
-		ArtifactPath: in.BuildID,
+		ArtifactPath: imageID,
+	}
+
+	// Default tag for the testplan image
+	defaultImageTag := fmt.Sprintf("%s:%s", in.TestPlan, imageID)
```

Comment:
If I build the `benchmarks` testplan, this would tag the image built in the previous step, such as:

```
REPOSITORY                                                                           TAG                                      IMAGE ID            CREATED              SIZE
<none>                                                                               <none>                                   aeaad624797f        About a minute ago   21.2MB
```

to

```
REPOSITORY                                                                           TAG                                      IMAGE ID            CREATED              SIZE
benchmarks                                                                           aeaad624797f                             aeaad624797f        About a minute ago   21.2MB
```